src/java.base/share/classes/java/text/Format.java

```diff
@@ -268,6 +268,17 @@ public Object clone() {
         }
     }
 
+    /**
+     * Used to distinguish JDK internal subclass and user-defined subclass
+     * of {code Format}.
+     *
+     * @return {@code true}  if current class is a JDK internal subclass of {code Format};
+     *         {@code false} otherwise
+     */
+    boolean isInternalSubclass() {
```

Comment:
Since this is defined in Format, can we apply similar changes of StringBuilder formatting to the other Format subclasses beyond just NumberFormat.

For example, in DateFormat, something such as,

```
<T extends Appendable & CharSequence> T formatWithGeneric(Date date,
                                                   T toAppendTo,
                                                   FieldPosition pos) {
        throw new UnsupportedOperationException("Subclasses should override this method");
}
```

Reply:
ok. I will update it if we have a conclusion about using  `StringBuf` or using `<T extends Appendable & CharSequence>`.

Reply:
This  comment inspire me that  use '<T extends Appendable & CharSequence>' in `SimpleDateFormat` is impossible, because the 'Appendable' lack of necessary `append` method which SimpleDateFormat requires.
So we need discuss whether `StringBuf` is a better solution?